### PR TITLE
Add test update_applicable_package_using_default_content_view

### PR DESCRIPTION
**Pull request by swadeley**
Hello

This adds a test for a package that is applicable for upgrade in Library, using Default Content View .
This test relates to the change to Pulp3.

Depends on: https://github.com/SatelliteQE/robottelo/pull/8619

### CODE REVIEW

File: tests/foreman/cli/test_errata.py [hunk 1]
```diff
@@ -1275,3 +1282,169 @@ def test_positive_check_errata_dates(module_org):
...
+    # python-psutil is obsoleted by python2-psutil, so install older python2-psutil for errata test
+    rhel77_contenthost_module.run(
+        'rpm -Uvh https://download-ib01.fedoraproject.org/pub/epel/7/'
+        'x86_64/Packages/p/python2-psutil-5.6.7-1.el7.x86_64.rpm'
+    )
+    rhel77_contenthost_module.install_katello_ca()
+    rhel77_contenthost_module.register_contenthost(module_manifest_org.label, new_module_ak.name)
+    host = entities.Host().search(query={'search': f'name={rhel77_contenthost_module.hostname}'})
+    host_id = host[0].id
+    host_content = entities.Host(id=host_id).read_json()
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_errata.py:1351]: you can use the `nailgun_host` property of the ContentHost object here as well

File: tests/foreman/cli/test_errata.py [hunk 2]
```diff
@@ -1275,3 +1282,169 @@ def test_positive_check_errata_dates(module_org):
...
+    # note time for later wait_for_tasks include 2 mins margin of safety.
+    timestamp = (datetime.utcnow() - timedelta(minutes=2)).strftime('%Y-%m-%d %H:%M')
+    # Update errata from Library, i.e. Default CV
+    chost.run(f'yum -y update --advisory {REAL_0_ERRATA_ID}')
+    # Wait for upload profile event (in case Satellite system slow)
+    host = entities.Host().search(query={'search': f'name={chost.hostname}'})
+    wait_for_tasks(
+        search_query=(
+            'label = Actions::Katello::Host::UploadProfiles'
+            f' and resource_id = {host[0].id}'
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_errata.py:1387]:
```suggestion
    wait_for_tasks(
        search_query=(
            'label = Actions::Katello::Host::UploadProfiles'
            f' and resource_id = {chost.nailgun_host.id}'
```

File: tests/foreman/cli/test_errata.py [hunk 3]
```diff
@@ -1275,3 +1282,169 @@ def test_positive_check_errata_dates(module_org):
...
+        2. Ensure the expected package is applicable on the newly registered host
+        3. Update the applicable package on the host
+        4. Ensure the package is no longer applicable
+
+    :expectedresults: package listed successfully and is installable
+
+    :CaseImportance: High
+    """
+    # get host ID
+    host = entities.Host().search(query={'search': f'name={chost.hostname}'})
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_errata.py:1416]: same `nailgun_host` property comment

File: tests/foreman/cli/test_errata.py [hunk 4]
```diff
@@ -1275,3 +1282,169 @@ def test_positive_check_errata_dates(module_org):
...
+    # note time for later wait_for_tasks include 2 mins margin of safety.
+    timestamp = (datetime.utcnow() - timedelta(minutes=2)).strftime('%Y-%m-%d %H:%M')
+    # Update package from Library, i.e. Default CV
+    chost.run(f'yum -y update {REAL_RHEL7_0_2_PACKAGE_NAME}')
+    # Wait for upload profile event (in case Satellite system slow)
+    host = entities.Host().search(query={'search': f'name={chost.hostname}'})
+    wait_for_tasks(
+        search_query=(
+            'label = Actions::Katello::Host::UploadProfiles'
+            f' and resource_id = {host[0].id}'
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_errata.py:1436]: same as above

File: tests/foreman/cli/test_errata.py [hunk 5]
```diff
@@ -1275,3 +1282,169 @@ def test_positive_check_errata_dates(module_org):
...
+            f' and started_at >= "{timestamp}"'
+        ),
+        search_rate=15,
+        max_tries=10,
+    )
+    # Assert that the package is no longer applicable
+    applicable_packages = Package.list(
+        {
+            'host-id': host[0].id,
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_errata.py:1445]: this would need to be adjusted as well

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: tests/foreman/cli/test_errata.py (hunk 1 above)
Comment (swadeley) [reply tests/foreman/cli/test_errata.py]: ok

File: tests/foreman/cli/test_errata.py (hunk 2 above)
Comment (swadeley) [reply tests/foreman/cli/test_errata.py]: done

File: tests/foreman/cli/test_errata.py (hunk 3 above)
Comment (swadeley) [reply tests/foreman/cli/test_errata.py:1414]: ok

File: tests/foreman/cli/test_errata.py (hunk 4 above)
Comment (swadeley) [reply tests/foreman/cli/test_errata.py]: done

File: tests/foreman/cli/test_errata.py (hunk 5 above)
Comment (swadeley) [reply tests/foreman/cli/test_errata.py]: done

File: tests/foreman/cli/test_errata.py [hunk 6]
```diff
@@ -1275,3 +1282,164 @@ def test_positive_check_errata_dates(module_org):
...
+    # python-psutil is obsoleted by python2-psutil, so install older python2-psutil for errata test
+    rhel77_contenthost_module.run(
+        'rpm -Uvh https://download-ib01.fedoraproject.org/pub/epel/7/'
+        'x86_64/Packages/p/python2-psutil-5.6.7-1.el7.x86_64.rpm'
+    )
+    rhel77_contenthost_module.install_katello_ca()
+    rhel77_contenthost_module.register_contenthost(module_manifest_org.label, new_module_ak.name)
+    host_id = rhel77_contenthost_module.nailgun_host.id
+    host_content = entities.Host(id=host_id).read_json()
+    assert host_content['subscription_status'] == 0
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_errata.py:1351]: Same question as your other pr

**Review by mshriver — APPROVED** [review]
Looks like all of Jake's comments have been addressed too.

**Comment by swadeley** [comment]
Thank you

**Pull request merged**